core/templates/dev/head/exploration_editor/ExplorationEditor.js

```diff
@@ -708,44 +727,41 @@ oppia.controller('ExplorationSaveAndPublishButtons', [
               };
 
               $scope.save = function() {
-                // If no objective has been specified yet, require the creator
-                // to specify it.
-                if ($scope.requireObjectiveToBeSpecified) {
-                  if (!explorationObjectiveService.displayed) {
-                    throw Error('Please specify an objective');
-                  }
-                  explorationObjectiveService.saveDisplayedValue();
-                  $scope.metadataList.push('objective');
+                // Throw an error if any values are invalid.
+                if (!explorationObjectiveService.displayed) {
+                  throw Error('Please specify an objective');
```

Comment:
Also nit: please make sure in both of these triplets of if-statements that they follow the conventional ordering for exploration metadata:

- title
- objective
- category